source/StepBro.Core/Execution/ScriptUtils.cs

```diff
@@ -222,7 +222,10 @@ public static string Await(this ILineReader reader, [Implicit] ICallContext cont
             reader.DebugDump();
 
             // If the reader has timestamp, set the timeout relative to the time of the current entry; otherwise just use current wall time.
-            DateTime entry = (reader.LinesHaveTimestamp && reader.Current != null) ? reader.Current.Timestamp : DateTime.Now;
+            // TODO: Setting the entry to the readers current lines timestamp, can in extreme cases be way behind, making the await be very inconsistent
+            //       for now we just use the current wall time until we find a way for the reader to use the correct entry time.
+            // DateTime entry = (reader.LinesHaveTimestamp && reader.Current != null) ? reader.Current.Timestamp : DateTime.Now;
+            DateTime entry = DateTime.Now;
```

Comment:
This is now a good fix. Imagine inspecting data that is minutes old, and you expect a certain line coming within 5 seconds after the "current" line, then that found line will still be minutes before current wall time.
It is better to insert Flush into the script, when lines from before current time should be discarded.

Reply:
How do you propose we determine whether lines from before current time should be discarded?